### PR TITLE
Handling set-cookie and removing some un-necessary code

### PR DESCRIPTION
**Pull request by shafeequ**
Currently, there is an issue when we have expiry time in the set-cookie and when http-library splits and joins it, it's breaking the semantics and hence these changes are required

### CODE REVIEW

File: src/HttpLibrary/__init__.py [hunk 1]
```diff
@@ -431,12 +431,14 @@ def get_response_header(self, header_name):
         keyword is a list containing both values.
         """
         self.response_should_have_header(header_name)
-        result = []
-        if self.get_major_python_version() < 3:
-            result = self.response.headers.getall(header_name)
+
+        # if there are multiple headers with same name, we need to return as a list
+        if header_name == 'Set-Cookie': # Need to be specific, as we dont know any other headers like this
```
Comment (surenderraju) [inline src/HttpLibrary/__init__.py]: Is it possible to add tests for these use-cases?

File: src/HttpLibrary/livetest.py [hunk 2]
```diff
@@ -227,4 +219,4 @@ def goto(self, href, method='get', **args):
         method = self.test_app.post
     return method(href, **args)
 
-webtest.TestResponse.goto = goto
+webtest.TestResponse.goto = goto
```
Comment (elbart) [inline src/HttpLibrary/livetest.py]: pls add the new line again, as github also indicates

File: src/HttpLibrary/__init__.py [hunk 3]
```diff
@@ -431,12 +431,14 @@ def get_response_header(self, header_name):
...
+            cookie_str = self.response.headers.getall(header_name)[0]
```
Comment (elbart) [inline src/HttpLibrary/__init__.py]: we only take the first result of get_all(). Could there be more? Would we need to iterate?

File: src/HttpLibrary/__init__.py [hunk 4]
```diff
@@ -431,12 +431,14 @@ def get_response_header(self, header_name):
...
+            cookies_list = cookie_str.split('auth_tkt')
```
Comment (elbart) [inline src/HttpLibrary/__init__.py]: i would prefer to make it explicit here with an if statement:
`if cookie_str.contains('auth_tkt'):`
only in this case, fiddle with `auth_tkt`

File: src/HttpLibrary/livetest.py (hunk 2 above)
Comment (shafeequ) [reply src/HttpLibrary/livetest.py]: sure, will add it

File: src/HttpLibrary/__init__.py (hunk 4 above)
Comment (shafeequ) [reply src/HttpLibrary/__init__.py]: auth_tkt will be there always hence did not add earlier but it does not harm if there is an additional if, so will incorporate the changes

**Pull request merged**